externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala

```diff
@@ -49,7 +49,11 @@ class TrinoSessionImpl(
     sessionManager: SessionManager)
   extends AbstractSession(protocol, user, password, ipAddress, conf, sessionManager) {
 
-  val sessionConf: KyuubiConf = sessionManager.getConf
+  val sessionConf: KyuubiConf = {
+    val engineConf = sessionManager.getConf
+    conf.foreach(kv => engineConf.set(kv._1, kv._2))
```

Comment:
Will this not pollute the engine conf?

Reply:
thanks for catch, using `sessionManager.getConf.clone` instead.

Reply:
@cxzl25 seems we lack a pre-assembled session conf in AbstractSession, we need some major refactor to simplify the current conf evaluation, I roughly remember you have raised a similar idea previously.